libraries/goclient/goLibrary.go

```diff
@@ -0,0 +1,126 @@
+package goclient
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/constants"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/host"
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/uds"
+)
+
+var (
+	hWR          uds.Handler
+	hPod         host.Handler
+	busy_timeout int = 20
+	busy_budget  int = 64
+)
+
+const (
+	requestDelay = 100 * time.Millisecond
+)
+
+/*
+Returns the version of our Handshake as a string
+*/
+func GetVersionStr() string {
+	return constants.Uds.Handshake.Version
+}
+
+/*
+Call this function to initialize the library, returns a cleanup function
+*/
+func Init() uds.CleanupFunc {
+	hWR = uds.NewHandler()
+	hPod = host.NewHandler()
+
+	// init uds Handler for reading and writing
+	if err := hWR.Init(constants.Uds.PodPath, constants.Uds.Protocol, constants.Uds.MsgBufSize, constants.Uds.CtlBufSize, 0*time.Second, ""); err != nil {
+		println("Library Error: Error Initialising UDS server: ", err)
+		os.Exit(1)
+	}
+
+	cleanup, err := hWR.Dial()
+	if err != nil {
+		println("Library Error: UDS Dial error:: ", err)
+		cleanup()
+		os.Exit(1)
+	}
+	hostname, err := hPod.Hostname()
+	if err != nil {
+		println("Library Error: Failed to initialize host: ", err)
+		os.Exit(1)
+	}
+	makeRequest("/connect, "+hostname, -1)
+	return cleanup
+}
+
+/*
+Returns the version of our Handshake
+*/
+func GetVersion() {
+	makeRequest("/version", -1)
+}
+
+/*
+Give it a list of device names and returns a map of the fds for each device and a cleanup function to close the connection
+*/
+func RequestXSKmapFD(devNames []string) map[string]int {
+	fds := make(map[string]int)
+
+	for _, dev := range devNames {
+		fd := makeRequest("/xsk_map_fd, "+dev, -1)
+		fds[dev] = fd
+	}
+
+	return fds
+}
+
+/*
+Specify a timeout limit, budget and fd to request a busy poll on the netdev
+*/
+func RequestBusyPoll(busyTimeout, busyBudget, fd int) int {
+	pollString := fmt.Sprintf("/config_busy_poll, %d, %d", busyTimeout, busyBudget)
+	pollFD := makeRequest(pollString, fd)
+	return pollFD
+}
+
+/*
+Logs an error with a message
+*/
+func logError(message string, e error) {
+	println(message, e)
+}
+
+/*
+Makes a request to the server
```

Comment:
So it might make sense to remove this makeRequest. We use a similar function in the e2e because it's only purpose is to print the handshake calls.

You're taking it step further here, your library functions should actually be returning the data within the responses - the FDs, the version, etc.

So you're going to need to update this function to return stuff.

However, it's getting a bit clunky. Each of your public functions are doing slightly different things, so it might make sense for them to be calling the Write() and Read() themselves.

Up to you.